tests/test_signals.py

```diff
@@ -0,0 +1,396 @@
+
+    @staticmethod
+    def calculate_total_steps(arguments_dict):
+        return arguments_dict['kwargs'].get('total_steps', UserTask.calculate_total_steps(arguments_dict))
+
+
+@shared_task(base=SampleTask, bind=True)
+def sample_task(self, user_id, argument, **kwargs):  # pylint: disable=unused-argument
+    print('Ran SampleTask for argument "{}"'.format(argument))
+    return argument
+
+
+@shared_task(base=SampleTask, bind=True)
+def missing_user_id(self, *args, **kwargs):  # pylint: disable=unused-argument
+    pass
+
+
+@shared_task(bind=True)
+def normal_task(self, *args, **kwargs):  # pylint: disable=unused-argument
+    return 'placeholder'
+
+
+class TestCreateUserTask(TestCase):
+    """
+    Tests of UserTaskStatus creation for new UserTasks.
+
+    Note that in Celery 3.1, sub-tasks of a group which are executed eagerly
+    (such as in many test suites) have no access to information about the
+    parent group.  Hence UserTaskStatus records cannot be automatically
+    created for the group as a whole in such tests (even though it works for
+    normal execution in worker processes).  This seems to be fixed in
+    Celery 4, which as of this writing has not yet been released.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user('test_user', 'test@example.com', 'password')
+
+    def test_create_user_task(self):
+        """The create_user_task signal handler should create a new UserTaskStatus record"""
+        result = sample_task.delay(self.user.id, 'Argument')
+        statuses = UserTaskStatus.objects.all()
+        assert len(statuses) == 1
+        status = statuses[0]
+        assert status.task_id == result.id
+        assert status.task_class == 'test_signals.sample_task'
+        assert status.user_id == self.user.id
+        assert status.parent is None
+        assert not status.is_container
+        assert status.name == 'SampleTask: Argument'
+        assert status.total_steps == 1
+        assert status.completed_steps == 0
+        assert status.state == UserTaskStatus.PENDING
+        assert status.attempts == 1
+
+    @override_settings(CELERY_ALWAYS_EAGER=True)
+    def test_create_user_task_eager(self):
+        """Eager tasks should still have UserTaskStatus records created on execution."""
+        result = sample_task.delay(self.user.id, 'Argument')
+        statuses = UserTaskStatus.objects.all()
+        assert len(statuses) == 1
+        status = statuses[0]
+        assert status.task_id == result.id
+        assert status.task_class == 'test_signals.sample_task'
+        assert status.user_id == self.user.id
+        assert status.parent is None
+        assert not status.is_container
+        assert status.name == 'SampleTask: Argument'
+        assert status.total_steps == 1
+        assert status.completed_steps == 1
+        assert status.state == UserTaskStatus.SUCCEEDED
+        assert status.attempts == 1
```

Comment:
Hmm...the trick there is needing to pass the value into the decorator.  `pytest-django` actually has a settings fixture that could be coupled with a parameterize decorator, but neither of them work in a `TestCase` subclass (and without that, `pytest-django` doesn't have an equivalent for `setUpTestData()`).  So have to choose between duplicating the test case or making two additional user insertions during the test run.  I could break out most of the assertions into a separate shared method, if that helps.